lib/vector/Vlib/copy.c

```diff
@@ -757,6 +771,8 @@ int Vect_copy_table_by_cats(struct Map_info *In, struct Map_info *Out,
     if (ret == -1) {
         G_warning(_("Unable to add database link for vector map <%s>"),
                   Out->name);
+        Vect_destroy_field_info(Fi);
```

Comment:
For `Vect_copy_table_by_cats()` let us fix this with goto labels, to limit repetition.

For this, we need two labels: `close_db_free_exit` and `free_exit` on either site of `db_close_database_shutdown_driver(driver);` at the end of the function. We also need to declare a `ret` int for return value.